### PR TITLE
DHFPROD-5437: Exclude hub artifact and entity model collections

### PR DESCRIPTION
**Pull request by rahulvudutala**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by briantang — APPROVED** [review]
Tested in the UI and works exactly as needed. Thanks!

**Review by ryanjdew — APPROVED** [review]

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs [hunk 1]
```diff
@@ -324,12 +324,26 @@ function convertStepReferenceToInlineStep(stepId) {
   return newFlowStep;
 }
 
+function getAllArtifactCollections() {
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs:327]: If we keep this feature, let's not add a function to this library when we only have one use for it.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs [hunk 2]
```diff
@@ -68,20 +69,29 @@ if(referenceType === 'element') {
 } else if(referenceType === 'field') {
   query = cts.fieldReference(propertyPath);
 } else if(referenceType === 'collection') {
-  query = cts.collectionReference();
+  const collectionsToIgnore = Artifacts.getAllArtifactCollections();
+  collectionsToIgnore.push("http://marklogic.com/entity-services/models", "hub-core-artifact");
+
+  const allCollectionsQuery = cts.collectionQuery(cts.collections());
+  const ignoredCollectionsQuery = queryObj.ignoreArtifactCollections ? cts.collectionQuery(collectionsToIgnore) : cts.collectionQuery([]);
+  query = cts.andNotQuery(allCollectionsQuery, ignoredCollectionsQuery);
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs:77]: Just do a cts.notQuery on a cts.collectionQuery that contains each of the collections you want to ignore. "cts.collections()" may be extremely slow when there are millions of collections.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs [hunk 3]
```diff
@@ -68,20 +69,29 @@ if(referenceType === 'element') {
...
-var facetValues = cts.valueMatch(query, pattern + "*",
-    ["item-order", "ascending", "limit=" + limit]).toArray().map(String);
-
-if (facetValues.length < limit) {
-  var moreFacetValues = cts.valueMatch(query, "?*" + pattern + "*",
-      ["item-order", "ascending", "limit=" + limit]).toArray().map(String);
-  facetValues = Array.from(
-      [...new Set([...facetValues, ...moreFacetValues])]).slice(0, limit);
+let facetValues = [];
+if(referenceType === 'collection') {
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs:84]: For better readability, put each block of code here into a separate function - e.g. findMatchingCollections and findMatchingIndexValues. Looks like there's some duplication between the two blocks of code to resolve as well.

**Review by rjrudin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**